model/host/hostutil.go

```diff
@@ -498,20 +500,75 @@ SeServiceLogonRight = $($currentSetting)
 `}, "\r\n"), nil
 }
 
+const passwordCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
+
+func generatePassword(length int) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = passwordCharset[rand.Int()%len(passwordCharset)]
+	}
+	return string(b)
+}
+
 // CreateServicePassword creates the password for the host's service user.
 func (h *Host) CreateServicePassword() error {
-	password := util.RandomString()
+	var password string
+	var valid bool
+	for i := 0; i < 1000; i++ {
+		password = generatePassword(12)
+		if strings.ToLower(password) == password {
```

Comment:
Is this check needed? We just need 3 of the 5 categories right?

Reply:
You're right. I removed this unnecessary check.